### PR TITLE
dbListTables now queries for view as well as table names

### PR DESCRIPTION
**Pull request by currymj**
Changes the query for listing tables as per #27.

### CODE REVIEW

**Comment by bosr** [comment]
Hi,

I also need the ability to query views. Could you merge it please?

Thank you!
Regards

File: R/tables.R [hunk 1]
```diff
@@ -112,8 +112,8 @@ setMethod("dbReadTable", c("PqConnection", "character"),
 #' @rdname postgres-tables
 setMethod("dbListTables", "PqConnection", function(conn) {
   dbGetQuery(conn, paste0(
-    "SELECT tablename FROM pg_tables WHERE schemaname !='information_schema'",
-    " AND schemaname !='pg_catalog'")
+    "SELECT table_name FROM INFORMATION_SCHEMA.tables ",
+    "WHERE table_schema = ANY (current_schemas(false))")
```
Comment (krlmlr) [inline R/tables.R]: Does this include temporary tables?

**Comment by msjgriffiths** [comment]
This query - 

```sql
SELECT table_name FROM INFORMATION_SCHEMA.tables WHERE table_schema = ANY (current_schemas(false));
```
.... only gets tables in the current `search_path`. This might be OK, but I'm not sure we should require users to have the `search_path` set correctly ahead of time. The current query, which excludes `information_schema` and `pg_catalog`, does not rely on the `search_path` being set; and pulls all schemas that are in the database (excluding the system tables). Reverting to the `search_path` might be seen as a regression.

For example, running that on a database I'm connecting to (Redshift), I get:

|   table_name      |
|-----------------------|
| table_taxonomy |

That's a real table, but it's only one of 33 real tables (it's the only table in the `public` schema, since `select current_schemas(false);` returns `{public}`).

I'd suggest instead:

```sql
SELECT table_name 
FROM INFORMATION_SCHEMA.tables 
WHERE table_schema not in ('information_schema', 'pg_catalog');
```

This produces:

|                table_name    |
|------------------------------|
| table_taxonomy|
| table1 |
| table2 |
| table3|
| ... |

(for all 33)

This should include **views**, **temporary tables**, and **tables** for all schema.

However, it might break if there are tables/views/etc with duplicate names.

**Comment by krlmlr** [comment]
I'd like to postpone this until schema support is available in DBI (rstats-db/DBI#24).

**Comment by krlmlr** [comment]
Thanks! Implemented a similar version in cc7a835f.

**Pull request closed without merging**